src/client/src/apps/SpotlightRoute/hooks.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  AutobahnConnection,
+  ConnectionEvent,
+  createConnection$,
+  RawServiceStatus,
+  retryWithBackOff, ServiceCollectionMap,
+  serviceStatusStream$,
+  ServiceStub,
+  ServiceStubWithLoadBalancer
+} from '../../rt-system'
+import { multicast, refCount, retryWhen } from 'rxjs/operators'
+import { ReplaySubject } from 'rxjs'
+import { useEffect, useState } from 'react'
+
+const HEARTBEAT_TIMEOUT = 3000;
+
+export function useServiceStub(autobahn: AutobahnConnection) {
+  const [serviceStub, setServiceStub] = useState(null)
+
+  useEffect(() => {
+    const connection$ = createConnection$(autobahn).pipe(
+      retryWhen(retryWithBackOff()),
+      multicast(() => {
+        return new ReplaySubject<ConnectionEvent>(1)
+      }),
+      refCount(),
+    )
+
+    const serviceStub = new ServiceStub('Spotlight', connection$)
+
+    const statusUpdates$ = serviceStub.subscribeToTopic<RawServiceStatus>('status')
+    const serviceStatus$ = serviceStatusStream$(statusUpdates$, HEARTBEAT_TIMEOUT).pipe(
+      multicast(() => {
+        return new ReplaySubject<ServiceCollectionMap>(1)
+      }),
+      refCount(),
+    )
+
+    const loadBalancedServiceStub = new ServiceStubWithLoadBalancer(serviceStub, serviceStatus$)
+
+    setServiceStub(loadBalancedServiceStub);
+  }, [])
```

Comment:
looks like `autobahn` should be added as a dependency. Do we ave linting for the react hooks dependencies setup?

Reply:
autobahn is imported from AutobahnConnection

Reply:
i mean it has to be a dependency of the hook